test/e2e/client/chowly-client.api.spec.ts

```diff
@@ -54,4 +47,110 @@ describe('ChowlyClient (e2e)', (): void => {
             expect(e.config['axios-retry'].retryCount).toBe(6);
         });
     }, 20000);
+
+    describe('#create', () => {
+        it('should not retry with status 501', async (): Promise<any> => {
+            const client = new ChowlyApiClient({
+                apiKey: 'test',
+                baseUrl: basePath,
+                maxRetries: undefined,
+            });
+            mock = new MockAdapter(client.client);
+            mock.onPost('/api/toos/v2/orders', chowlyOrder).reply(501);
+            await client.create(chowlyOrder).catch((e) => {
```

Comment:
Can we have a consistent pattern?
`Promise.catch` or `try-catch`

Reply:
`Promise.catch` will be the solution here, please update all tests where you use `try-catch`